09-maze_solver/progress/internal/solver/solver.go

```diff
@@ -50,28 +52,54 @@ func (s *Solver) Solve() error {
 
 	go func() {
 		// The first pixel is on the edge, the second pixel is inwards.
-		s.pathsToExplore <- []image.Point{entrance, {1, entrance.Y}}
+		s.pathsToExplore <- &path{previousStep: nil, at: entrance}
 	}()
 
-	// Launch the goroutine in charge of drawing the GIF image.
-	go s.drawFrames()
+	wg := sync.WaitGroup{}
+	wg.Add(2)
 
-	// Listen for new paths to explore. This only returns when the maze is solved.
-	s.listenToBranches()
+	go func() {
+		defer wg.Done()
+		// Launch the goroutine in charge of drawing the GIF image.
+		s.drawFrames()
+	}()
+
+	go func() {
+		defer wg.Done()
+		// Listen for new paths to explore. This only returns when the maze is solved.
+		s.listenToBranches()
+	}()
+
+	wg.Wait()
+
+	s.finalise()
 
 	return nil
 }
 
 // findEntrance returns the position of the maze entrance on the image.
 func (s *Solver) findEntrance() (image.Point, error) {
-	height := s.maze.Bounds().Dy()
-
-	// We built the maze with the entrance on the left border, and not in a corner
-	for y := 1; y < height-1; y++ {
-		if s.maze.RGBAAt(0, y) == s.config.entranceColour {
-			return image.Point{0, y}, nil
+	for row := 0; row < s.maze.Bounds().Dy(); row++ {
```

Comment:
Are we removing a constraint?

Reply:
Per https://github.com/alienorlatour/tiny-go-projects/pull/75#discussion_r1342178130